tests/sync/app.cpp

```diff
@@ -285,6 +631,70 @@ TEST_CASE("app: login_with_credentials unit_tests", "[sync][app]") {
         CHECK(processed);
     }
 }
+*/
+
+TEST_CASE("app: UserAPIKeyProviderClient unit_tests", "[sync][app]") {
+    
+    auto setup_user = []() {
+        if (realm::SyncManager::shared().get_current_user()) {
+            return;
+        }
+        realm::SyncUserIdentifier identifier {
+            UnitTestTransport::user_id,
+            UnitTestTransport::auth_route
+        };
+        realm::SyncManager::shared().get_user(identifier,
+                                              good_access_token,
+                                              good_access_token);
+    };
+    
+    std::unique_ptr<GenericNetworkTransport> (*factory)() = []{
+        return std::unique_ptr<GenericNetworkTransport>(new UnitTestTransport);
+    };
+    
+    auto config = App::Config{"translate-utwuv", factory};
+    auto app = App(config);
+    try_make_dir("./" + config.app_id);
+    SyncManager::shared().configure({ .base_file_path = "./" + config.app_id });
+    bool processed = false;
+    ObjectId obj_id(UnitTestTransport::api_key_id.c_str());
+
+    SECTION("create api key") {
+        setup_user();
+        app.provider_client<App::UserAPIKeyProviderClient>().create_api_key(UnitTestTransport::api_key_name,
+                                                                            [&](Optional<App::UserAPIKey> user_api_key, Optional<AppError> error) {
+            CHECK(!error);
+            CHECK(user_api_key->disabled == false);
+            CHECK(user_api_key->id.to_string() == UnitTestTransport::api_key_id);
+            CHECK(user_api_key->key == UnitTestTransport::api_key);
+            CHECK(user_api_key->name == UnitTestTransport::api_key_name);
+        });        
+    }
+    
+    SECTION("fetch api key") {
```

Comment:
After adjusting the setup code above to properly clean up state between tests, this test and the ones below fail because it never actually creates a user.

Reply:
I'm now calling `setup_user();` between each test for that scenario.